Fix most of test-webkitpy on Python 3.12

gsnedders opened the pull request:
#### 46e33611bc5cbc47e8baa50a2cad8b2c2225f1bd
<pre>
Fix most of test-webkitpy on Python 3.12
<a href="https://bugs.webkit.org/show_bug.cgi?id=260877">https://bugs.webkit.org/show_bug.cgi?id=260877</a>

Reviewed by Jonathan Bedard.

This bumps many dependencies to versions which support Python 3.12,
largely either because of meta hooks changes (six, also vendored via
urllib3), because of removals (pylint, soupsieve), or because of use of
non-public API (pyfakefs).

This downgrades the version of urllib3 we are declaring on Python 3.12
as we are unable to install &gt;=2 due to the lack of support of PEP 517
(see <a href="https://bugs.webkit.org/show_bug.cgi?id=261082).">https://bugs.webkit.org/show_bug.cgi?id=261082).</a> This also has the
advantage of using the same version on all configurations.

Additionally, change both versions of requests (again, to a singular
version) to be compatible with the version of urllib3 we&apos;re specifying.

Also, remove Genshi, because as noted in
<a href="https://bugs.webkit.org/show_bug.cgi?id=261103">https://bugs.webkit.org/show_bug.cgi?id=261103</a> it is in fact
unused. This avoids having to deal with its incompatibilities.

Finally, when running via pytest, install everything via the
AutoInstaller prior to the start of test execution

* Tools/Scripts/libraries/webkitcorepy/webkitcorepy/__init__.py:
* Tools/Scripts/webkitpy/__init__.py:
* Tools/Scripts/webkitpy/conftest.py:
(pytest_collection_modifyitems):
(pytest_collection_finish):

Canonical link: <a href="https://commits.webkit.org/268856@main">https://commits.webkit.org/268856@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/70c1d682b22c8f6e86f94acf88f9bef5d84e69d5

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/20788 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/21196 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/21855 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/22678 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/19384 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/21024 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/24439 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/21375 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/20691 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/21009 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/20795 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/18057 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/23531 "Built successfully") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/20974 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/17960 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/18869 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/25175 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/19039 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/19047 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/23076 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/19634 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/16679 "Passed tests") | | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/20691 "Passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/18867 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/5007 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/23198 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/19452 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

gsnedders commented on Tools/Scripts/webkitpy/__init__.py:
```diff
@@ -35,7 +35,7 @@
 
 if sys.version_info >= (3, 7):
-    AutoInstall.register(Package('pylint', Version(2, 6, 0)))
+    AutoInstall.register(Package('pylint', Version(2, 13, 9)))
```
This may not suffice:

```
pylint 2.13.9 requires astroid, which is not installed.
pylint 2.13.9 requires dill, which is not installed.
pylint 2.13.9 requires isort, which is not installed.
pylint 2.13.9 requires mccabe, which is not installed.
pylint 2.13.9 requires platformdirs, which is not installed.
```

gsnedders replied:
These are all used only by optional checkers, so let's not pull in all of this now.

JonWBedard reviewed (approved):
r=me. This change looks to work on all the configurations we're currently defending, and moving us towards working on Python 3.12 is always a good thing.

gsnedders committed:
Fix most of test-webkitpy on Python 3.12
https://bugs.webkit.org/show_bug.cgi?id=260877

Reviewed by Jonathan Bedard.

This bumps many dependencies to versions which support Python 3.12,
largely either because of meta hooks changes (six, also vendored via
urllib3), because of removals (pylint, soupsieve), or because of use of
non-public API (pyfakefs).

This downgrades the version of urllib3 we are declaring on Python 3.12
as we are unable to install >=2 due to the lack of support of PEP 517
(see https://bugs.webkit.org/show_bug.cgi?id=261082). This also has the
advantage of using the same version on all configurations.

Additionally, change both versions of requests (again, to a singular
version) to be compatible with the version of urllib3 we're specifying.

Also, remove Genshi, because as noted in
https://bugs.webkit.org/show_bug.cgi?id=261103 it is in fact
unused. This avoids having to deal with its incompatibilities.

Finally, when running via pytest, install everything via the
AutoInstaller prior to the start of test execution

* Tools/Scripts/libraries/webkitcorepy/webkitcorepy/__init__.py:
* Tools/Scripts/webkitpy/__init__.py:
* Tools/Scripts/webkitpy/conftest.py:
(pytest_collection_modifyitems):
(pytest_collection_finish):

Canonical link: https://commits.webkit.org/268856@main
